common/src/main/java/io/netty/util/concurrent/AutoScalingEventExecutorChooserFactory.java

```diff
@@ -277,6 +280,27 @@ public void run() {
                     return;
                 }
 
+                // First, ensure our state accurately reflects reality before making any decisions.
+                // This handles cases where an executor was woken up externally since the last check.
+                final AutoScalingState oldState = state.get();
+                final AutoScalingState currentState = rebuildActiveExecutors();
+                final Set<EventExecutor> newlyWokenExecutors = new HashSet<>();
+
+                if (oldState.activeChildrenCount != currentState.activeChildrenCount) {
+                    final Set<EventExecutor> oldActive = new HashSet<>(Arrays.asList(oldState.activeExecutors));
+
+                    for (EventExecutor currentActive : currentState.activeExecutors) {
+                        if (!oldActive.contains(currentActive)) {
```

Comment:
Also maybe we not even need the `Set` at all as these really shouldn't be too many anyway and so looping might just be fine to find ?